packages/flutter/lib/src/gestures/mouse_tracking.dart

```diff
@@ -0,0 +1,308 @@
+  /// updated, right after rendering.
+  void collectMousePositions() {
+    void exitAllAnnotations() {
+      for (_TrackedAnnotation tracked in _trackedAnnotations) {
+        if (tracked.active) {
+          if (tracked.annotation?.onExit != null) {
+            tracked.annotation.onExit(MouseExitDetails(
+              globalPosition: lastMouseEvent?.position,
+              sourceTimeStamp: lastMouseEvent?.timeStamp,
+            ));
+          }
+          tracked.active = false;
+        }
+      }
+    }
+
+    // This indicates that a mouse pointer was removed, or has not been
+    // connected yet. If no mouse is connected, then we want to make sure that
+    // all active annotations are exited.
+    if (!mouseIsConnected) {
+      exitAllAnnotations();
+      return;
+    }
+
+    final Set<_TrackedAnnotation> notified = Set<_TrackedAnnotation>();
+    final MouseDetectorAnnotation hit = annotationFinder(lastMouseEvent.position);
+
+    // No annotation was found at this position, so send an exit to all active
+    // tracked annotations, since none of them were hit.
+    if (hit == null) {
+      exitAllAnnotations();
+      return;
+    }
+
+    final _TrackedAnnotation tracked = _findAnnotation(hit);
+    if (tracked.active) {
+      // A tracked annotation that is already active, and needs to have its
+      // location updated.
+      if (tracked.annotation?.onMove != null) {
+        tracked.annotation.onMove(MouseMoveDetails(
+          globalPosition: lastMouseEvent.position,
+          sourceTimeStamp: lastMouseEvent.timeStamp,
+        ));
+      }
+    } else {
+      // A tracked annotation that just became active and needs to have enter
+      // and move events sent to it.
+      tracked.active = true;
+      if (tracked.annotation?.onEnter != null) {
+        tracked.annotation.onEnter(MouseEnterDetails(
+          globalPosition: lastMouseEvent.position,
+          sourceTimeStamp: lastMouseEvent.timeStamp,
+        ));
+      }
+      if (tracked.annotation?.onMove != null) {
+        tracked.annotation.onMove(MouseMoveDetails(
+          globalPosition: lastMouseEvent.position,
+          sourceTimeStamp: lastMouseEvent.timeStamp,
+        ));
+      }
+    }
+    notified.add(tracked);
+
+    // Tell any tracked annotations that weren't hit that they are not longer
+    // active.
+    for (_TrackedAnnotation tracked in _trackedAnnotations.difference(notified)) {
+      if (tracked.active) {
+        if (tracked.annotation?.onExit != null) {
+          tracked.annotation.onExit(MouseExitDetails(
+            globalPosition: lastMouseEvent.position,
+            sourceTimeStamp: lastMouseEvent.timeStamp,
+          ));
+        }
+        tracked.active = false;
+      }
+    }
+  }
+
+  /// The most recent mouse event observed. May be null if no mouse is
+  /// connected, or hasn't produced an event yet. Will not be updated unless
+  /// there is at least one tracked annotation.
+  PointerEvent lastMouseEvent;
```

Comment:
should this be private? or at a minimum, a private member with a public getter but no public setter?

Reply:
Yes, it should be private.